pkg/kubelet/prober/results/results_manager.go

```diff
@@ -80,7 +79,7 @@ func (r Result) ToPrometheusType() float64 {
 type Update struct {
 	ContainerID kubecontainer.ContainerID
 	Result      Result
-	PodUID      types.UID
+	Pod         *v1.Pod
```

Comment:
Why do we need to switch to passing pod via pointer instead of by UID?

Reply:
I'm afraid that this will race with the `podManager` since both doesn't seem to be in sync, and they can "observe" different states at the same time https://github.com/kubernetes/kubernetes/pull/114994#discussion_r1080650751

Reply:
/hold while i integrate this in my head and figure out whether 113145 fixes this / limits this

Reply:
After poking at this for a few hours, I think we have a fundamental accident in the Kubelet.

Pod manager is upstream of the pod worker - loops that work on the "desired state of the kubelet" should never consult pod manager.  The pod manager we have today is really "configPodManager" or "desiredPodManager".  The pod worker really needs to own a "workingPodManager" or "actualPodManager", and then subsystems that use configPodManager *all* need to be using workingPodManager.  I.e. whether a pod is force deleted in config is irrelevant - only whether the pod worker has accepted it matters.  This affects volume manager, status manager, stat manager, eviction manager, and probably a few other places in the Kubelet.

I'm going to take a stab at what a possible change would look like and see what the blast radius is.  But, as a consequence we will likely fix a bunch of bugs we don't realize we have in a lot of wierd corner cases with both force deletion and static pods.